### ISSUE TITLE
Using node-gyp compiled modules in sm engine

### ISSUE THREAD

**Issue by darrrk** [jxcore/jxcore#510]
There is any way to use node-gyp compiled modules in sm engine? If no, would it be in future and how?

**Comment by obastemur** [comment]
ATM leveldown (under leveldown-mobile) is SM compatible. Which module exactly and to what platform ?

**Comment by darrrk** [comment]
Platform: OS X 10.10.5

When I try `jx install`, next errors happened:

```
> fsevents@0.3.8 install /Users/darky/Desktop/tavmant/node_modules/gulp-watch/node_modules/chokidar/node_modules/fsevents
> node-gyp rebuild

Target arch x64
  SOLINK_MODULE(target) Release/.node
  SOLINK_MODULE(target) Release/.node: Finished
  CXX(target) Release/obj.target/fse/fsevents.o
In file included from ../fsevents.cc:6:
In file included from ../node_modules/nan/nan.h:24:
In file included from /Users/darky/.node-gyp/jxb304/src/node.h:45:
In file included from /Users/darky/.node-gyp/jxb304/src/jx/Proxy/JSEngine.h:55:
In file included from /Users/darky/.node-gyp/jxb304/src/jx/Proxy/Mozilla/MozJS/MozJS.h:6:
In file included from /Users/darky/.node-gyp/jxb304/src/jx/Proxy/Mozilla/MozJS/Isolate.h:17:
In file included from /Users/darky/.node-gyp/jxb304/src/jsapi.h:12:
In file included from /Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:12:
In file included from /Users/darky/.node-gyp/jxb304/src/mozilla/Assertions.h:16:
In file included from /Users/darky/.node-gyp/jxb304/src/mozilla/Attributes.h:12:
In file included from /Users/darky/.node-gyp/jxb304/src/mozilla/Compiler.h:12:
/Users/darky/.node-gyp/jxb304/src/mozilla/Char16.h:173:15: error: unknown type
      name 'char16_t'
typedef const char16_t* char16ptr_t;
              ^
In file included from ../fsevents.cc:6:
In file included from ../node_modules/nan/nan.h:24:
In file included from /Users/darky/.node-gyp/jxb304/src/node.h:45:
In file included from /Users/darky/.node-gyp/jxb304/src/jx/Proxy/JSEngine.h:55:
In file included from /Users/darky/.node-gyp/jxb304/src/jx/Proxy/Mozilla/MozJS/MozJS.h:6:
In file included from /Users/darky/.node-gyp/jxb304/src/jx/Proxy/Mozilla/MozJS/Isolate.h:17:
In file included from /Users/darky/.node-gyp/jxb304/src/jsapi.h:12:
In file included from /Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:12:
In file included from /Users/darky/.node-gyp/jxb304/src/mozilla/Assertions.h:318:
/Users/darky/.node-gyp/jxb304/src/mozilla/TypeTraits.h:90:36: error: use of
      undeclared identifier 'char16_t'
template<> struct IsIntegralHelper<char16_t>           : TrueType {};
                                   ^
/Users/darky/.node-gyp/jxb304/src/mozilla/TypeTraits.h:213:27: warning: rvalue
      references are a C++11 extension [-Wc++11-extensions]
struct IsRvalueReference<T&&> : TrueType {};
                          ^
/Users/darky/.node-gyp/jxb304/src/mozilla/TypeTraits.h:354:25: error: use of
      undeclared identifier 'char16_t'
template<> struct IsPod<char16_t>           : TrueType {};
                        ^
/Users/darky/.node-gyp/jxb304/src/mozilla/TypeTraits.h:714:25: warning: rvalue
      references are a C++11 extension [-Wc++11-extensions]
struct RemoveReference<T&&>
                        ^
In file included from ../fsevents.cc:6:
In file included from ../node_modules/nan/nan.h:24:
In file included from /Users/darky/.node-gyp/jxb304/src/node.h:45:
In file included from /Users/darky/.node-gyp/jxb304/src/jx/Proxy/JSEngine.h:55:
In file included from /Users/darky/.node-gyp/jxb304/src/jx/Proxy/Mozilla/MozJS/MozJS.h:6:
In file included from /Users/darky/.node-gyp/jxb304/src/jx/Proxy/Mozilla/MozJS/Isolate.h:17:
In file included from /Users/darky/.node-gyp/jxb304/src/jsapi.h:12:
In file included from /Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:12:
/Users/darky/.node-gyp/jxb304/src/mozilla/Assertions.h:328:30: warning:
      variadic templates are a C++11 extension [-Wc++11-extensions]
template<typename R, typename... A>
                             ^
In file included from ../fsevents.cc:6:
In file included from ../node_modules/nan/nan.h:24:
In file included from /Users/darky/.node-gyp/jxb304/src/node.h:45:
In file included from /Users/darky/.node-gyp/jxb304/src/jx/Proxy/JSEngine.h:55:
In file included from /Users/darky/.node-gyp/jxb304/src/jx/Proxy/Mozilla/MozJS/MozJS.h:6:
In file included from /Users/darky/.node-gyp/jxb304/src/jx/Proxy/Mozilla/MozJS/Isolate.h:17:
In file included from /Users/darky/.node-gyp/jxb304/src/jsapi.h:12:
In file included from /Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:14:
/Users/darky/.node-gyp/jxb304/src/mozilla/Casting.h:31:3: error: use of
      undeclared identifier 'static_assert'; did you mean 'static_cast'?
  static_assert(sizeof(From) == sizeof(To),
  ^
In file included from ../fsevents.cc:6:
In file included from ../node_modules/nan/nan.h:24:
In file included from /Users/darky/.node-gyp/jxb304/src/node.h:45:
In file included from /Users/darky/.node-gyp/jxb304/src/jx/Proxy/JSEngine.h:55:
In file included from /Users/darky/.node-gyp/jxb304/src/jx/Proxy/Mozilla/MozJS/MozJS.h:6:
In file included from /Users/darky/.node-gyp/jxb304/src/jx/Proxy/Mozilla/MozJS/Isolate.h:17:
In file included from /Users/darky/.node-gyp/jxb304/src/jsapi.h:12:
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:97:33: error:
      expected ')'
  static_assert((Base::kSignBit & Base::kExponentBits) == 0,
                                ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:97:17: note: to match
      this '('
  static_assert((Base::kSignBit & Base::kExponentBits) == 0,
                ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:97:24: error: C++
      requires a type specifier for all declarations
  static_assert((Base::kSignBit & Base::kExponentBits) == 0,
                       ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:97:24: error:
      parameter declarator cannot be qualified
  static_assert((Base::kSignBit & Base::kExponentBits) == 0,
                 ~~~~~~^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:97:56: error:
      expected ')'
  static_assert((Base::kSignBit & Base::kExponentBits) == 0,
                                                       ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:97:16: note: to match
      this '('
  static_assert((Base::kSignBit & Base::kExponentBits) == 0,
               ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:97:3: error: C++
      requires a type specifier for all declarations
  static_assert((Base::kSignBit & Base::kExponentBits) == 0,
  ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:99:33: error:
      expected ')'
  static_assert((Base::kSignBit & Base::kSignificandBits) == 0,
                                ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:99:17: note: to match
      this '('
  static_assert((Base::kSignBit & Base::kSignificandBits) == 0,
                ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:99:24: error: C++
      requires a type specifier for all declarations
  static_assert((Base::kSignBit & Base::kSignificandBits) == 0,
                       ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:99:24: error:
      parameter declarator cannot be qualified
  static_assert((Base::kSignBit & Base::kSignificandBits) == 0,
                 ~~~~~~^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:99:59: error:
      expected ')'
  static_assert((Base::kSignBit & Base::kSignificandBits) == 0,
                                                          ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:99:16: note: to match
      this '('
  static_assert((Base::kSignBit & Base::kSignificandBits) == 0,
               ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:99:3: error: C++
      requires a type specifier for all declarations
  static_assert((Base::kSignBit & Base::kSignificandBits) == 0,
  ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:101:38: error:
      expected ')'
  static_assert((Base::kExponentBits & Base::kSignificandBits) == 0,
                                     ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:101:17: note: to match
      this '('
  static_assert((Base::kExponentBits & Base::kSignificandBits) == 0,
                ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:101:24: error: C++
      requires a type specifier for all declarations
  static_assert((Base::kExponentBits & Base::kSignificandBits) == 0,
                       ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:101:24: error:
      parameter declarator cannot be qualified
  static_assert((Base::kExponentBits & Base::kSignificandBits) == 0,
                 ~~~~~~^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:101:64: error:
      expected ')'
  static_assert((Base::kExponentBits & Base::kSignificandBits) == 0,
                                                               ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:101:16: note: to match
      this '('
  static_assert((Base::kExponentBits & Base::kSignificandBits) == 0,
               ^
/Users/darky/.node-gyp/jxb304/src/mozilla/FloatingPoint.h:101:3: error: C++
      requires a type specifier for all declarations
  static_assert((Base::kExponentBits & Base::kSignificandBits) == 0,
  ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
3 warnings and 20 errors generated.
make: *** [Release/obj.target/fse/fsevents.o] Error 1
gyp ERR! build error
gyp ERR! stack onExit@/Users/darky/.jx/npm/node_modules/node-gyp/lib/build.js:271:23
gyp ERR! stack emit@events.js:85:7
gyp ERR! stack ChildProcess/this._handle.onexit@child_process.js:824:7
gyp ERR! stack
gyp ERR! System Darwin 14.5.0
gyp ERR! command "jx" "/Users/darky/.jx/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/darky/Desktop/tavmant/node_modules/gulp-watch/node_modules/chokidar/node_modules/fsevents
gyp ERR! node -v v0.10.40
gyp ERR! node-gyp -v v0.13.1
gyp ERR! not ok
npm WARN optional dep failed, continuing fsevents@0.3.8
```

Later, when I try to run my gulp-based project:

```
TypeError: result.match(...) is null (/Users/darky/Desktop/tavmant/node_modules/gulp/node_modules/v8flags/index.js 63:16)
    at getFlags/< (/Users/darky/Desktop/tavmant/node_modules/gulp/node_modules/v8flags/index.js:63:17)
    at exithandler (child_process.js:663:7)
    at emit (events.js:85:7)
    at maybeClose (child_process.js:771:5)
    at ChildProcess.prototype.spawn/</< (child_process.js:994:11)
    at emit (events.js:82:7)
    at Socket.prototype._destroy/< (net.js:424:7)
```

**Comment by ktrzeciaknubisa** [comment]
Are you doing it with SpiderMonkey? Did you try with V8 version?

**Comment by darrrk** [comment]
Yes, it with SpiderMonkey. With V8 all ok.

**Comment by darrrk** [comment]
I think main problem on this issue #551
But here confused errors above, when `jx install`

**Comment by darrrk** [comment]
Closed in favor #691
